catalog/transformer/catalog-transformer-csv-common/src/main/java/ddf/catalog/transformer/csv/common/CsvTransformer.java

```diff
@@ -0,0 +1,178 @@
+      printColumnHeaders(csvPrinter, sortedAttributeDescriptors, aliasMap);
+
+      metacards.forEach(m -> printMetacard(csvPrinter, m, sortedAttributeDescriptors));
+
+      return csvPrinter.getOut();
+    } catch (IOException ioe) {
+      throw new CatalogTransformerException(ioe);
+    }
+  }
+
+  private static void printMetacard(
+      final CSVPrinter csvPrinter,
+      final Metacard metacard,
+      final List<AttributeDescriptor> sortedAttributeDescriptors) {
+
+    Iterator<Serializable> metacardIterator =
+        new MetacardIterator(metacard, sortedAttributeDescriptors);
+
+    printData(csvPrinter, metacardIterator);
+  }
+
+  private static void printColumnHeaders(
+      final CSVPrinter csvPrinter,
+      final List<AttributeDescriptor> sortedAttributeDescriptors,
+      final Map<String, String> aliasMap) {
+    Iterator<String> columnHeaderIterator =
+        new ColumnHeaderIterator(sortedAttributeDescriptors, aliasMap);
+
+    printData(csvPrinter, columnHeaderIterator);
+  }
+
+  private static void printData(final CSVPrinter csvPrinter, final Iterator iterator) {
+    try {
+      csvPrinter.printRecord(() -> iterator);
+    } catch (IOException ioe) {
+      LOGGER.debug("Failed to print the CSV data.", ioe);
+    }
+  }
+
+  public static List<AttributeDescriptor> sortAttributes(
+      final Set<AttributeDescriptor> attributeSet, final List<String> attributeOrder) {
+    CsvAttributeDescriptorComparator attributeComparator =
+        new CsvAttributeDescriptorComparator(attributeOrder);
+
+    return attributeSet.stream().sorted(attributeComparator).collect(toList());
+  }
+
+  /**
+   * Given a list of {@link Metacard}s and a string set of hiddenFields, returns a set of {@link
+   * AttributeDescriptor}s that contains all attributes that exist on the given metacard types
+   * except those that should be excluded.
+   */
+  public static Set<AttributeDescriptor> getAllAttributes(
+      final List<Metacard> metacards, final Set<String> excludedFields) {
+
+    final Predicate<AttributeDescriptor> attributeNotBinary =
+        attrDesc ->
+            !AttributeType.AttributeFormat.BINARY.equals(attrDesc.getType().getAttributeFormat());
+
+    final Predicate<AttributeDescriptor> attributeNotObject =
+        attrDesc ->
+            !AttributeType.AttributeFormat.OBJECT.equals(attrDesc.getType().getAttributeFormat());
+
+    final Predicate<AttributeDescriptor> attributeNotExcluded =
+        attrDesc -> !excludedFields.contains(attrDesc.getName());
+
+    return metacards
+        .stream()
+        .filter(Objects::nonNull)
+        .map(Metacard::getMetacardType)
+        .map(MetacardType::getAttributeDescriptors)
+        .flatMap(Set::stream)
+        .filter(attributeNotBinary)
+        .filter(attributeNotObject)
+        .filter(attributeNotExcluded)
+        .collect(toSet());
+  }
+
+  /**
+   * Given a list of {@link Metacard}s, a string set of requested attributes and a string set of
```

Comment:
:pencil2: This three-item list should use [the Oxford comma](https://www.grammarly.com/blog/what-is-the-oxford-comma-and-why-do-people-care-so-much-about-it/)!